File: src/tmp/PassTest.java
public class PassTest {

    public static void main(String[] args) {
        // Prints "Hello, World" to the terminal window.
        System.out.println("Hello, World");
        Ball ball = new Ball(10);
        System.out.println(ball.size);
        setSize(ball, 100);
        System.out.println(ball.size);
        setBall(ball);
        System.out.println(ball.size);
    }

    static class Ball{
        public int size = 1;
        Ball(int s){
            size = s;
        }
    }
    // Object reference are passed by value
    static void setSize(Ball b, int s){
        b.size = s;
    }
    // java is pass by value
    static void setBall(Ball b){
        Ball bb = new Ball(7);
        b = bb;
    }
}

File: src/tmp/PassTest.cpp
#include <iostream> 
using namespace std; 


// pass by value
void swap1(int a, int b){
    int tmp = a;
    a = b;
    b = tmp;
}
// pass by pointer (address)
void swap2(int *a, int *b){
    int tmp = *a;
    *a = *b;
    *b = tmp;
}
// pass by reference
void swap3(int &a, int &b){
    int tmp = a;
    a = b;
    b = tmp;
}

int main() { 
   cout << "Hello! World!" << endl;

   int a = 1;
   int b = 100;
   cout << "a:" << a << " b:" << b << endl;
   swap1(a, b);
   a = 1;
   b = 100;
   cout << "a:" << a << " b:" << b << endl;
   swap2(&a, &b);
   cout << "a:" << a << " b:" << b << endl;
   a = 1;
   b = 100;
   swap3(a,b);
   cout << "a:" << a << " b:" << b << endl;
   
 return 0; 
}

